apps/web/src/components/GalleryEditor/PiecesSidebar/SidebarWalletSelector.tsx

```diff
@@ -0,0 +1,118 @@
+import { useCallback, useMemo, useState } from 'react';
+import { graphql, useFragment } from 'react-relay';
+import styled from 'styled-components';
+
+import { Dropdown } from '~/components/core/Dropdown/Dropdown';
+import { DropdownItem } from '~/components/core/Dropdown/DropdownItem';
+import { DropdownSection } from '~/components/core/Dropdown/DropdownSection';
+import IconContainer from '~/components/core/IconContainer';
+import { HStack } from '~/components/core/Spacer/Stack';
+import { BaseM } from '~/components/core/Text/Text';
+import { ChainMetadata } from '~/components/GalleryEditor/PiecesSidebar/chains';
+import { chainsMap } from '~/components/GalleryEditor/PiecesSidebar/chains';
+import { SidebarWalletSelectorFragment$key } from '~/generated/SidebarWalletSelectorFragment.graphql';
+import DoubleArrowsIcon from '~/icons/DoubleArrowsIcon';
+import { useTrack } from '~/shared/contexts/AnalyticsContext';
+import { removeNullValues } from '~/shared/relay/removeNullValues';
+
+export type SidebarWallet = { chainAddress: { chain: string; address: string } } | 'All';
+
+type SidebarWalletSelectorProps = {
+  queryRef: SidebarWalletSelectorFragment$key;
+  selectedChain: ChainMetadata;
+  selectedWallet: SidebarWallet;
+  onSelectedWalletChange: (selectedWallet: SidebarWallet) => void;
+};
+
+export default function SidebarWalletSelector({
+  queryRef,
+  selectedChain,
+  selectedWallet,
+  onSelectedWalletChange,
+}: SidebarWalletSelectorProps) {
+  const [isDropdownOpen, setIsDropdownOpen] = useState(false);
+  const { viewer } = useFragment(
+    graphql`
+      fragment SidebarWalletSelectorFragment on Query {
+        viewer {
+          ... on Viewer {
+            user {
+              wallets {
+                chainAddress @required(action: THROW) {
+                  address @required(action: THROW)
+                  chain @required(action: THROW)
+                }
+              }
+            }
+          }
+        }
+      }
+    `,
+    queryRef
+  );
+  const userWalletsOnSelectedNetwork = useMemo(
+    () =>
+      removeNullValues(viewer?.user?.wallets).filter((wallet) => {
+        if (wallet.chainAddress.chain === chainsMap[selectedChain.name].baseChain) {
+          return wallet;
+        }
+      }),
+    [viewer?.user?.wallets, selectedChain]
+  );
+
+  const track = useTrack();
+
+  const onSelectWallet = useCallback(
```

Comment:
very minor nit: handlers defined on the component render should start with `handleX` instead of `onX`; usually `onX` is reserved for names of props